ratelimit/backends/simple.py

```diff
@@ -0,0 +1,108 @@
+import asyncio
+import time
+from collections import defaultdict
+from threading import Lock
+from typing import Dict, List, Optional
+
+from ..rule import Rule
+from . import BaseBackend
+
+lock = Lock()
+
+
+def synchronized(func):
+    def wrapper(*args, **kwargs):
+        try:
+            lock.acquire()
+            return func(*args, **kwargs)
+        finally:
+            lock.release()
+
+    return wrapper
+
+
+class MemoryBackend(BaseBackend):
+    """simple limiter with memory"""
+
+    def __init__(self):
+        # user: deadline
+        self.blocked_users: Dict[str, int] = {}
+        # path: {rule_key: (limit, timestamp)}
+        self.blocks: Dict[str, Dict[str, List[int]]] = defaultdict(dict)
+
+    @staticmethod
+    def now() -> int:
+        return int(time.time())
```

Comment:
https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.time is better in the case.